### PR TITLE
Add the new handbook

### PR DESCRIPTION
**Pull request by hakusaro**
Hiya, I'm Lucas. I'm proposing a series of formal policies to replace our old framework for operating the TShock project & associated repositories.

At a glance, this handbook:

* Attempts to clarify the role of leaders.
* Attempts to clarify how we make decisions as leaders.
* Attempts to create a set of unified policy guidelines we can follow in the future.
* **Formally resets permissions for all leaders to the new baseline.**
  * If (and when) this pull request is merged, the associated members list will become the official policy and set of members. This structure is _implied_ as of now, but we want to make it _explicit_ or _explicitly revert to a more conservative policy_ with this PR too.

Currently done:

- [x] Membership document
- [x] Consensus document
- [x] Trust document
- [x] Readme

Currently todo:

- [x] Policy folder with policy on banning
- [x] A micro-RFC process to get policy clarification with consensus
- [ ] A technical guide to updating TShock

I want feedback from everyone on this. I really do aim to get everyone, but if we can't get everyone, we're going to follow the guidance I have in ```consensus.md``` to try and get this policy implemented by Tuesday, August 22, 2017.

### CODE REVIEW

**Comment by hakusaro** [comment]
I want people to read these policies as well as they can and use interpretation of things. It's not a legal document -- but I want it to be clear on our intentions and meanings.

I do have some specific scenarios, though, that everyone should try to answer as a comment to validate that the policy works.

* If I go MIA for two months, who is in charge?
* Who can push code to ```general-devel``` on TShock and when can they?
* Who can create new official TShock releases, and when?
* What happens if someone like george goes inactive for a month and we can't contact them?
* If someone gets banned by the group and appeals their ban with a fair point, what do we do?
* What can Zaicon do compared to what MarioE can do?
* What can ProfessorX do compared to what Enerdy can do?

Please answer these questions when you have time, @Pryaxis/tshock. If the policy works, we should get the same answers from everyone and not just me. Since we're trying to reach a uniform guide everyone understands, I'll probably heavily revise the policy I've built to accommodate help normalize the responses.

**Comment by hakusaro** [comment]
RT

**Comment by QuiCM** [comment]
>Shank MIA

Ijwu and I

>Push to `gen-dev`

TShock contributors who've previously created high quality, reviewed pull requests. I.E., trusted code contributors. Major changes should follow the 2 man rule & have discussion from key contributors and knowledge holders

>Official releases

Shank, Ijwu, and I

>George goes awol

Nothing, unless the stat server crashes. In which case we hire a shady individual to find him and hit him with a plank

>Ban appeal w/ fair point

Group discussion

>Zaicon v MarioE

Zaicon's focus thus far has been user management, MarioE's has been code

>Enerdy v ProfX

Both are trash tier :>
Again though, Prof has been more user focused

**Comment by hakusaro** [comment]
> In which case we hire a shady individual to find him and hit him with a plank

lol

**Comment by Patrikkk-zz** [comment]
```If I go MIA for two months, who is in charge?```
Ijwu and White

```Who can push code to general-devel on TShock and when can they?```
Any of the leaders with write access "can". Tho they must go through the 2 man rule (unless it's an emergency and a "higher"/older member such as White/Ijwu etc thinks it is necessary). So simppy put when; after discussion with an other member.

```Who can create new official TShock releases, and when?```
Currently Shank, White and Ijwut

```What happens if someone like george goes inactive for a month and we can't contact them?```
We panic that stats are down and noone can put it back up? We pray they didn't get hit by a bus?
On a serious note, i assume this refers whether to keep them in the group if they disappear without notice and 0 contact?

```If someone gets banned by the group and appeals their ban with a fair point, what do we do?``` 
Discuss it.

```What can Zaicon do compared to what MarioE can do?```
Since question asks what they "can" do, answer is: the same things. Same access.

```What can ProfessorX do compared to what Enerdy can do?```
Same answer as above.

**Comment by hakusaro** [comment]
@Patrikkk 
> On a serious note, i assume this refers whether to keep them in the group if they disappear without notice and 0 contact?

Dead on.

**Comment by hakusaro** [comment]
@Pryaxis/tshock I'm done with what I consider the base policy to be. I want comments and feedback on what is or isn't clear, as well as your answers to the scenario. I'll be answering questions and revising the documents so that it's clear what everything means.

This policy framework will go into effect based on consensus data from code review on this PR. If you disagree or want changes you must **request changes** or else we're going to assume you just don't care. If you don't approve or deny, you're also in the same camp. You must **make a decision**.

**Review by Patrikkk-zz — APPROVED** [review]
All looks good. No questions i have on mind at the moment. :) if so, I'll open an issue later on for discussion.

**Comment by hakusaro** [comment]
@Patrikkk I accidentally your approval can you try that again?

**Review by QuiCM — APPROVED** [review]
LGTM

**Comment by AxisKriel** [comment]
```Who can push code to general-devel on TShock and when can they?```
All leaders, but flow is prefered over directly pushing code to general-devel whenever possible, excluding very tiny fixes that don't change the codebase much.

```Who can create new official TShock releases, and when?```
Those with root access (Shank, White and Ijwu), with leaders possibly taking over in case of absense. The timing for a new release is usually decided based on the amount of changed pushed to general-devel since the last release (perhaps this should be explicit on the handbook?)

```What happens if someone like george goes inactive for a month and we can't contact them?```
Generally, we can temporarily remove them from the leader's seat until their situation is clarified. In George's case, however, a backup plan should be arranged seeing as he's our data manager and the one who keeps the Stats server running. More people should probably get access to it in some way.

```If someone gets banned by the group and appeals their ban with a fair point, what do we do?```
Despite going back on bans in the past, we're meant to put more thought on our bans as of this handbook's implementation. The policy is now to never go back on ban decisions, or at least not without a very good reason, seeing as contradictions weaken said policy.

```What can Zaicon do compared to what MarioE can do?```
As of this document, they have the same power, regardless of their acomplishments in the past; while Zaicon contributed in the form of community management, MarioE's major contributions were made in code for our projects.

```What can ProfessorX do compared to what Enerdy can do?```
Same as above, we are both leaders.

**Review by Patrikkk-zz — APPROVED** [review]
"Accidentally" all good.  👍 No questions for the moment.

**Comment by mistzzt** [comment]
> If I go MIA for two months, who is in charge?

Ijwu and White
>Who can push code to general-devel on TShock and when can they?

Leaders with "write" access can push to this branch. They can only push commits when a pull request is approved according to two-man rule, or something has been discussed.
> Who can create new official TShock releases, and when?

Root users can create new release generally when some important update created.
> What happens if someone like george goes inactive for a month and we can't contact them?

A backup plan should be discussed for stats server.
> If someone gets banned by the group and appeals their ban with a fair point, what do we do?

Discuss it.
> What can Zaicon do compared to what MarioE can do?

According to `members.md`, _Leaders of Pryaxis are the people who run the operation. Every leader is equivalent._
> What can ProfessorX do compared to what Enerdy can do?

Same

**Review by Ijwu — APPROVED** [review]

**Comment by DaBomberNTG** [comment]
> If I go MIA for two months, who is in charge?

Ijwu & White
> Who can push code to general-devel on TShock and when can they?

Leaders with write access can push code to general-dev & they can push after it's been reviewed by at least 1 other leaders, i.e. follow the 2 man rule

> Who can create new official TShock releases, and when?

Root users, and can do so there is such a release to be released!
> What happens if someone like george goes inactive for a month and we can't contact them?

~~Shut it _all_ down~~ A temporary removal of position as long as their hiatus lasts and figure out what to do next after they return based on the situation. In terms of george specifically, if it's feasible a think a 2nd in command for the stats server wouldn't be a bad idea.

> If someone gets banned by the group and appeals their ban with a fair point, what do we do?

Discussion! 
> What can Zaicon do compared to what MarioE can do?

I 2nd @Patrikkk 's answers for this question & the question below; they all have the same access. Although they can all do the same, @QuiCM has listed the various focuses these 4 individuals have had and would be inclined to say "stick to what you know/like" 
> What can ProfessorX do compared to what Enerdy can do?

Read previous answer.

**Comment by hakusaro** [comment]
@Enerdy please approve or deny this pull request.
@mistzzt please approve or deny this pull request.

**Comment by hakusaro** [comment]
@Zaicon please approve or deny this pull request.

**Review by AxisKriel — APPROVED** [review]

**Review by DaBomberNTG — APPROVED** [review]

**Review by mistzzt — APPROVED** [review]
lgtm

**Comment by hakusaro** [comment]
- [ ] history
- [ ] current/future projects list/agenda

**Comment by hakusaro** [comment]
- [ ] l/i of llc

**Review by Celant — APPROVED** [review]
:+1:

**Comment by hakusaro** [comment]
@zaicon @professorxz?

**Comment by Cassie1996** [comment]
Possibly adding contact details or emails of each person for ease of contact. The addition of protocols for when a specific situation should arise is a good thing. 
A good introduction for users to show how they may support the group via patreon or donation links like PayPal.

**Review by Zaicon — APPROVED** [review]
LGTM

**Review by cheahjs — APPROVED** [review]

**Review by SignatureBeef — APPROVED** [review]
`If I go MIA for two months, who is in charge?`
The other root members - White & Ijwu

`Who can push code to general-devel on TShock and when can they?`
Root users & Leaders, typically following the two man rule. 

`Who can create new official TShock releases, and when?`
Root users, when there have been updates/improvements that have had adequate testing.

`What happens if someone like george goes inactive for a month and we can't contact them?`
It's not clear if Root users are able to take over to fix any issues, or even if they have access. If not then this is not an ideal situation.
Should something terrible happen we could probably roll out a drop-in replacement stat tracker. (Just as an example anyway, i'm not in the loop with any of the services...and it's madness to do this)

`If someone gets banned by the group and appeals their ban with a fair point, what do we do?`
Ideally we should not be in this situation if they clearly conflict with the handbook - we should not be lifting a ban after repetitively warning the user anyway.
However, discussing it with the group is always good.

`What can Zaicon do compared to what MarioE can do?`
`What can ProfessorX do compared to what Enerdy can do?`
As per the handbook they are all leaders, however what they specialize in may vary. Maybe these roles could be outlined as to what the leader prefers and/or is familiar with? e.g. forums, tshock, tsapi, plugin submissions

With that out of the way, I'm pretty happy with the direction of this PR thus far.

**Comment by ivanbiljan** [comment]
`If I go MIA for two months, who is in charge?`
The root members - White & Ijwu

`Who can push code to general-devel on TShock and when can they?`
Root users & Leaders.

`Who can create new official TShock releases, and when?`
The root members again.

`What happens if someone like george goes inactive for a month and we can't contact them?`
See Death's answer.

`If someone gets banned by the group and appeals their ban with a fair point, what do we do?`
Discuss whether the ban should be lifted or not. As pretty much everyone else said.

`What can Zaicon do compared to what MarioE can do?`
What can ProfessorX do compared to what Enerdy can do?
I'm not sure what you are asking here. What are our permissions as of now? How capable are we of doing something? What each of us should have access to? 
As others have already stated, in Mario v Zaicon's case each of them helped this community in some way. Zaicon has been primarily focused on user management & support, but he has also helped with updating TSAPI before we switched when noone else had time to, while Mario has been making useful code contributions for the last 4-5 years.
As for me and Enerdy, I don't even know how to answer this. Enerdy was somewhat active on the forums before and made code contributions to Pryaxis projects. I on the other hand was in charge of helping Zaicon with spam cleaning, and since then have made contributions to TShock and Orion which is why I was given push access to TShock's repo by White.

**Comment by hakusaro** [comment]
> As for me and Enerdy, I don't even know how to answer this. Enerdy was somewhat active on the forums before and made code contributions to Pryaxis projects. I on the other hand was in charge of helping Zaicon with spam cleaning, and since then have made contributions to TShock and Orion which is why I was given push access to TShock's repo by White.

No, you're all equal.

> As per the handbook they are all leaders, however what they specialize in may vary. Maybe these roles could be outlined as to what the leader prefers and/or is familiar with? e.g. forums, tshock, tsapi, plugin submissions

Leader discretion or unofficial tasking or whatever they want to do. See when Enerdy opted out of doing plugin approvals, for example.

> I 2nd @Patrikkk 's answers for this question & the question below; they all have the same access. Although they can all do the same, @QuiCM has listed the various focuses these 4 individuals have had and would be inclined to say "stick to what you know/like"

Stick to what you know/like/want to do. Or do something else. No biggie.

> As of this document, they have the same power, regardless of their acomplishments in the past;

Dead on.

**Comment by hakusaro** [comment]
@ProfessorXZ also please approve/deny if you're okay/not okay.

**Review by ivanbiljan — APPROVED** [review]

**Comment by Cassie1996** [comment]
Possibly adding a contact detail or email for users to suggest new ideas or changes to help improve policies or detail work.

**Comment by Cassie1996** [comment]
+activity scheduler for members as a implementable and viable system for leaders to set a time in their busy schedule to meet and discuss issues or ideas put forward on the table.

Or/and

+To-do lists for members to implement agendas set forth by founder to make sure on going progress with set policies or ideas in motion which is also maintained by contanst updates by members flagging WIP/on-hold/delayed/aborted/complete this method allows proper facilitated planning of set agendas where each member knows what role he/she has to play a specific role instead of a generic coding function.

**Comment by Ijwu** [comment]
> Contact Details

I like this. At least for the root users.

> +members

I don't believe we have an inherent obligation to the community. Let me clarify, since that's a broad statement. Besides being decent humans and facilitating polite and civil communication, Pryaxis doesn't inherently owe the community anything. We *want* to provide a server mod and we do love being available and doing this work. We aren't obligated to continue our services at any point, though.

We have the problem sometimes that community members feel we owe them something. I don't believe we do. I know that's not what you said directly, but if it makes it into the handbook that we are around to 
> provide the quality and user-friendly support when necessary so as to achieve improved T-shock knowledge

then this contradicts my impression.

> +activity scheduler

See my previous point. Leaders can do this voluntarily but I don't think it's appropriate to hit the handbook.

> +To-do lists

I feel pretty meh about this. Either we do stuff or we don't. We aren't yet structured as a work organization and, frankly, I don't want my hobby turning into work. I will do what I need to for Pryaxis, I don't need project management around this in a way that gives me work flashbacks.

> +ban appeal

This deserves some discussion. I don't want a ban appeal in the same way many people have one. That being allowing banned users to appeal their ban because they just want to get unbanned, despite being guilty of what they did to get banned.

The ban appeal should be to give members a chance to disprove that they did what got them banned.

> +discord details and community expectations

Discord details: yes. Community expectations: we have rules that lay out everything we expect.

**Pull request merged**
